src/java.base/share/man/java.md

```diff
@@ -2883,6 +2883,48 @@ Java HotSpot VM.
     expensive operations. Using a lower value will cause heap memory to be
     uncommitted earlier, at the risk of soon having to commit it again.
 
+`-XX:+UseShenandoahGC`
+:   Enables the use of the Shenandoah garbage collector. This is a low pause
+    time, concurrent garbage collector. Its pause times are not proportional to
+    the size of the heap. Shenandoah garbage collector can work with compressed
+    pointers. See `-XX:UseCompressedOops` for further information about
+    compressed oops.
+
+`-XX:ShenandoahGCMode=`*mode*
+:   Sets the GC mode for Shenandoah GC to use. By default, this option is set
+    to `satb`. Among other things, this defines which barriers are in use.
+    Possible mode values include the following:
+
+    `satb`
+    :   Snapshot-at-the-beginning concurrent GC (three pass mark-evac-update).
+
+    `passive`
+    :   Stop the world GC only (either degenerated or full). This mode is
+        diagnostic, and must be enabled via `-XX:+UnlockDiagnosticVMOptions`.
```

Comment:
I don't think we add diagnostic options to the generic options list. These are not for end users, and we do not surface them unnecessarily.

Reply:
Will remove